packages/devtools_app/lib/src/screens/memory/area_controls/controls_area.dart

```diff
@@ -0,0 +1,681 @@
+            '${analytics_constants.memoryDisplayInterval}-$newValue',
+          );
+          _controller.displayInterval = chartInterval(newValue!);
+          final duration = chartDuration(_controller.displayInterval);
+
+          widget.eventChartController.zoomDuration = duration;
+          widget.vmChartController.zoomDuration = duration;
+          widget.androidChartController.zoomDuration = duration;
+        });
+      },
+      items: _displayTypes,
+    );
+  }
+
+  Widget _memorySourceDropdown(TextTheme textTheme) {
+    final files = _controller.memoryLog.offlineFiles();
+
+    // Can we display dropdowns in verbose mode?
+    final isVerbose =
+        _controller.memorySourcePrefix == _memorySourceMenuItemPrefix;
+
+    // First item is 'Live Feed', then followed by memory log filenames.
+    files.insert(0, MemoryController.liveFeed);
+
+    final allMemorySources = files.map<DropdownMenuItem<String>>((
+      String value,
+    ) {
+      // If narrow width compact the displayed name (remove prefix 'memory_log_').
+      final displayValue =
+          (!isVerbose && value.startsWith(MemoryController.logFilenamePrefix))
+              ? value.substring(MemoryController.logFilenamePrefix.length)
+              : value;
+      return SourceDropdownMenuItem<String>(
+        value: value,
+        child: Text(
+          '${_controller.memorySourcePrefix}$displayValue',
+          key: sourcesKey,
+        ),
+      );
+    }).toList();
+
+    return RoundedDropDownButton<String>(
+      key: sourcesDropdownKey,
+      isDense: true,
+      style: textTheme.bodyText2,
+      value: _controller.memorySource,
+      onChanged: (String? newValue) {
+        setState(() {
+          ga.select(
+            analytics_constants.memory,
+            analytics_constants.sourcesDropDown,
+          );
+          _controller.memorySource = newValue!;
+        });
+      },
+      items: allMemorySources,
+    );
+  }
+
+  Widget _createToggleAdbMemoryButton() {
+    return IconLabelButton(
+      icon: _controller.isAndroidChartVisible ? Icons.close : Icons.show_chart,
+      label: 'Android Memory',
+      onPressed: _isAndroidCollection
+          ? _controller.toggleAndroidChartVisibility
+          : null,
+      minScreenWidthForTextBeforeScaling: 900,
+    );
+  }
+
+  Widget _buildMemoryControls(TextTheme textTheme) {
+    final mediaWidth = MediaQuery.of(context).size.width;
+    _controller.memorySourcePrefix = mediaWidth > _verboseDropDownMinimumWidth
+        ? _memorySourceMenuItemPrefix
+        : '';
+
+    return Row(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        _memorySourceDropdown(textTheme),
+        const SizedBox(width: defaultSpacing),
+        if (_controller.isConnectedDeviceAndroid ||
+            _controller.isOfflineAndAndroidData)
+          _createToggleAdbMemoryButton(),
+        const SizedBox(width: denseSpacing),
+        _isAdvancedSettingsEnabled
+            ? Row(
```

Comment:
wrap this Row in a ValueListenableBuilder that listens to `_controller.advancedSettingsEnabled`, and then remove the call to `addAutoDisposeListener` for this listenable above in `didChangeDependencies`. We do not need to rebuild the entire widget tree based on this value, but instead just this Row. 

It is best practice to push state changes farther down the widget tree so that we are not unnecessarily rebuilding things that do not need to be rebuilt.